web_poet/exceptions.py

```diff
@@ -0,0 +1,29 @@
+class ReloadResponseData(Exception):
+    """Indicates that there's something wrong with :class:`~.ResponseData` and
+    it needs to be downloaded again.
+
+    You should use this in instance where the HTTP Response is malformed, missing
+    some required information, etc.
+
+    This should be raised inside the subclasses of :class:`~.WebPage` or anything
+    which requires :class:`~.ResponseData` as a dependency. The framework which
+    handles the Page Objects should then attempt to create a new instance with a
+    fresh :class:`~.ResponseData` dependency.
+    """
+
+    pass
+
+
+class DelegateFallback(Exception):
```

Comment:
should be split into _expected_ and _unexpected_ fallback requests. One of them should trigger some sort of alerts or monitoring actions _(e.g. unexpected layouts)_.